staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -328,52 +363,56 @@ func (p *jsonPatcher) applyPatchToCurrentObject(currentObject runtime.Object) (r
 	return objToUpdate, nil
 }
 
-func (p *jsonPatcher) createNewObject() (runtime.Object, error) {
+func (p *jsonPatcher) createNewObject(_ context.Context) (runtime.Object, error) {
 	return nil, errors.NewNotFound(p.resource.GroupResource(), p.name)
 }
 
 // applyJSPatch applies the patch. Input and output objects must both have
 // the external version, since that is what the patch must have been constructed against.
-func (p *jsonPatcher) applyJSPatch(versionedJS []byte) (patchedJS []byte, retErr error) {
+func (p *jsonPatcher) applyJSPatch(versionedJS []byte) (patchedJS []byte, strictErrors []error, retErr error) {
 	switch p.patchType {
 	case types.JSONPatchType:
 		// sanity check potentially abusive patches
 		// TODO(liggitt): drop this once golang json parser limits stack depth (https://github.com/golang/go/issues/31789)
-		if len(p.patchBytes) > 1024*1024 {
+		if len(p.patchBytes) > 1024*1024 || p.validationDirective == metav1.FieldValidationStrict || p.validationDirective == metav1.FieldValidationWarn {
 			v := []interface{}{}
-			if err := json.Unmarshal(p.patchBytes, &v); err != nil {
-				return nil, errors.NewBadRequest(fmt.Sprintf("error decoding patch: %v", err))
+			var err error
+			if strictErrors, err = kjson.UnmarshalStrict(p.patchBytes, &v); err != nil {
+				return nil, nil, errors.NewBadRequest(fmt.Sprintf("error decoding patch: %v", err))
 			}
 		}
 
 		patchObj, err := jsonpatch.DecodePatch(p.patchBytes)
```

Comment:
`jsonpatch.DecodePatch` won't decode `p.patchBytes` strictly, will it? this is where we'd need to detect duplicates or unknown fields in the patch like this:
`[{"op":"foo","op":"bar","unknown":"baz"}]`

seems like we need better test coverage of unknown or duplicate fields in a json patch

Reply:
Done, tests added. @apelisse suggested we distinguish between strict errors coming from the json patch ops vs the Decode into, so I've prepended "json patch" to these errors. lemme know what you think about this.